modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IO.kt

```diff
@@ -1108,4 +1097,18 @@ fun <A> IOOf<A>.handleError(f: (Throwable) -> A): IO<A> =
  * @see handleErrorWith for a version that can resolve the error using an effect
  */
 fun <A> IOOf<A>.handleErrorWith(f: (Throwable) -> IOOf<A>): IO<A> =
-  IO.Bind(fix(), IOFrame.Companion.ErrorHandler(f))
+  BIO.Bind(fix(), IOFrame.Companion.ErrorHandler(f))
+
+/**
+ * Executes the given `finalizer` when the source is finished, either in success or in error, or if canceled, allowing
+ * for differentiating between exit conditions. That's thanks to the [ExitCase] argument of the finalizer.
+ *
+ * As best practice, it's not a good idea to release resources via `guaranteeCase` in polymorphic code.
```

Comment:
It would be good to say here why is not a good idea, as reader unfamiliar with Bracket, they will be left wondering.

Reply:
Yes, the documentation keep use a refresher. I'm hoping to go over all snippets in the final PR which where we'll need to tackle all broken docs.. 😭